benches/odoh_bench.rs

```diff
@@ -1,61 +1,59 @@
 use criterion::{black_box, criterion_group, criterion_main, Criterion};
-use hpke::Serializable;
-use odoh_rs::key_utils::*;
-use odoh_rs::protocol::*;
+use odoh_rs::*;
+use rand::{rngs::StdRng, SeedableRng};
+use std::convert::TryInto;
 
-fn generate_key_pair() -> ObliviousDoHKeyPair {
-    let ikm = "871389a8727130974e3eb3ee528d440a871389a8727130974e3eb3ee528d440a";
-    let ikm_bytes = hex::decode(ikm).unwrap();
-    let (secret_key, public_key) = derive_keypair_from_seed(&ikm_bytes);
-    let public_key_bytes = public_key.to_bytes().to_vec();
-    let odoh_public_key = ObliviousDoHConfigContents {
-        kem_id: 0x0020,
-        kdf_id: 0x0001,
-        aead_id: 0x0001,
-        public_key: public_key_bytes,
-    };
-    ObliviousDoHKeyPair {
-        private_key: secret_key,
-        public_key: odoh_public_key,
-    }
-}
+pub fn bench_steps(c: &mut Criterion) {
+    let mut rng = StdRng::from_seed([0; 32]);
+    let key_pair = ObliviousDoHKeyPair::new(&mut rng);
 
-fn generate_query_body() -> ObliviousDoHQueryBody {
-    ObliviousDoHQueryBody::new(
-        &hex::decode("5513010000010000000000000377777706676f6f676c6503636f6d00001c0001").unwrap(),
-        None,
-    )
-}
+    c.bench_function("step_encrypt_query", |b| {
+        b.iter(|| {
```

Comment:
You can move input generation to `b.iter_batched(|| input() |input| ... bench ...)` to avoid benchmarking that.

Reply:
Practically the interfaces be called after data deserializationc, so putting it into the bench can give a general idea of total cost.